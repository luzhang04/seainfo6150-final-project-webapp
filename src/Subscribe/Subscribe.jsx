import React, {useState} from "react";
import styles from "./Subscribe.module.css";

const Subscribe = () => {
  const [submittedForm, setSubmittedForm] = useState();

    const [userEmail, setUserEmail] = useState();

	function onSubmit(event) {
		event.preventDefault();
		const data = new FormData(event.target);
		setSubmittedForm(data);
	};

  return (
    <div className = {styles.container}>
  <h1> Subscribe for DIY updates</h1>
  {
    submittedForm ? (
      <div><h2 className={styles.success}>Success, {submittedForm.get("firstname")}! </h2>
      <h2>we'll send updates to {submittedForm.get("email")} at the frequency of {submittedForm.get("dropdown")}!</h2></div>
    ) : (
      <form onSubmit={onSubmit} className = {styles.content}>
          <div className="container">
              <p>Please fill in this form to subscribe, fields with * are required.</p>


              <label htmlFor="firstname"><b>*First Name: </b></label>
              <div><input id = "firstname" type="text" placeholder="Firstname" name="firstname" required></input></div>
              <br/>


              <label htmlFor="lastname"><b>*Last Name: </b></label>
              <div><input id = "lastname" type="text" placeholder="Lastame" name="lastname" required></input></div>
              <br/>

              <label htmlFor="email"><b>*Email: </b></label>
              <div><input id = "email" type="text" placeholder="Email" pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}$" title="Enter a valid email address." name="email" required></input></div>
              <br/>

              <label htmlFor="dropdownId"><b>How frequent would you like to hear from us? </b></label>
              <div>
            <select name="dropdown" id="dropdownId">
                <option value="everyday">Every Day</option>
                <option value="everyweek">Every Week</option>
                <option value="everymonth">Every Month</option>
             </select>
               </div>
              <br/>

              <label htmlFor="tel"><b>Phone Number: </b> </label>
              <div><input id = "tel" type="text" placeholder="000-000-0000" name="tel" pattern="[0-9]{10}" title="Phone number is in ten-digit format"></input></div>
              <br/>

              <div><b>Would you like to receive text messages from us? </b></div>
              <input type="radio" id="yes" name="radio" value="yes"></input>
              <label htmlFor="yes">Yes</label><br/>
              <input type="radio" id="no" name="radio" value="no"></input>
              <label htmlFor="no">No</label>
              <br/>
              <br/>

              <div><b>What crafts are you interested in? (please select all)</b></div>
              <input type="checkbox" name="checkbox1" value="EasterDIY"></input>
              <label htmlFor="checkbox1"> Easter DIY </label><br />
              <input id = "checkbox1" type="checkbox" name="checkbox2" value="Valentine's-Day-diy"></input>
              <label htmlFor="checkbox2"> Valentine DIY</label><br/>
              <input id = "checkbox2" type="checkbox" name="checkbox3" value="Christmas-diy"></input>
              <label htmlFor="checkbox3"> Christmas DIY</label><br/>
              <input id = "checkbox3" type="checkbox" name="checkbox4" value="Halloween-diy"></input>
              <label htmlFor="checkbox3"> Halloween DIY</label>
              <br/>
              <br/>

              <label htmlFor="dropdownId2"><b>How did you hear about us? </b></label>
              <div>
            <select name="dropdown2" id="dropdownId2">
                <option value="--">----</option>
                <option value="SearchEngine">Search engine</option>
                <option value="GoogleAds">Google Ads</option>
                <option value="YoutubeAds">Youtube Ads</option>
                <option value="Twitter post">Twitter post</option>
                <option value="Radio">Radio</option>
                <option value="TV">TV</option>
                <option value="Newspaper">Newspaper</option>
                <option value="Others">Others</option>
             </select>
               </div>
               <br/>
               <br/>
              <button type="submit"><b>Subscribe</b></button>

          </div>

      </form>
    )
  }
</div>
  )
}

export default Subscribe;
