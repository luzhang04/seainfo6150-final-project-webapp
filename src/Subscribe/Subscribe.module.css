.container {
  background:#edf6f9;
  border: 1px solid #edf6f9;
  align-items: center;
  gap: 20px;
  display:flex;
  flex-direction: column;
  font-family: 'Montserrat', Helvetica, sans-serif;
  list-style-type: none;
  padding: 0 80px;
  justify-items: center;
  height:100%;
}

.content {

  padding: 10px;
}

.success {
  text-align: center;
}

b {
  font-size: 1.3em;
}
