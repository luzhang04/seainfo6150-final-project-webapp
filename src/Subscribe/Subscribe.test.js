import React from "react";
import { render } from "@testing-library/react";
import Subscribe from "./Subscribe.jsx";

describe("Subscribe tests", () => {
    it("renders correctly", () => {
      const { container } = render(<Subscribe />);
      expect(container).toMatchSnapshot();
    });
  });