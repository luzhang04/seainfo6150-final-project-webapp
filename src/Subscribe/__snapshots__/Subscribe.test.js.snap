// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscribe tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
       Subscribe for DIY updates
    </h1>
    <form
      class="content"
    >
      <div
        class="container"
      >
        <p>
          Please fill in this form to subscribe, fields with * are required.
        </p>
        <label
          for="firstname"
        >
          <b>
            *First Name: 
          </b>
        </label>
        <div>
          <input
            id="firstname"
            name="firstname"
            placeholder="Firstname"
            required=""
            type="text"
          />
        </div>
        <br />
        <label
          for="lastname"
        >
          <b>
            *Last Name: 
          </b>
        </label>
        <div>
          <input
            id="lastname"
            name="lastname"
            placeholder="Lastame"
            required=""
            type="text"
          />
        </div>
        <br />
        <label
          for="email"
        >
          <b>
            *Email: 
          </b>
        </label>
        <div>
          <input
            id="email"
            name="email"
            pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,}$"
            placeholder="Email"
            required=""
            title="Enter a valid email address."
            type="text"
          />
        </div>
        <br />
        <label
          for="dropdownId"
        >
          <b>
            How frequent would you like to hear from us? 
          </b>
        </label>
        <div>
          <select
            id="dropdownId"
            name="dropdown"
          >
            <option
              value="everyday"
            >
              Every Day
            </option>
            <option
              value="everyweek"
            >
              Every Week
            </option>
            <option
              value="everymonth"
            >
              Every Month
            </option>
          </select>
        </div>
        <br />
        <label
          for="tel"
        >
          <b>
            Phone Number: 
          </b>
           
        </label>
        <div>
          <input
            id="tel"
            name="tel"
            pattern="[0-9]{10}"
            placeholder="000-000-0000"
            title="Phone number is in ten-digit format"
            type="text"
          />
        </div>
        <br />
        <div>
          <b>
            Would you like to receive text messages from us? 
          </b>
        </div>
        <input
          id="yes"
          name="radio"
          type="radio"
          value="yes"
        />
        <label
          for="yes"
        >
          Yes
        </label>
        <br />
        <input
          id="no"
          name="radio"
          type="radio"
          value="no"
        />
        <label
          for="no"
        >
          No
        </label>
        <br />
        <br />
        <div>
          <b>
            What crafts are you interested in? (please select all)
          </b>
        </div>
        <input
          name="checkbox1"
          type="checkbox"
          value="EasterDIY"
        />
        <label
          for="checkbox1"
        >
           Easter DIY 
        </label>
        <br />
        <input
          id="checkbox1"
          name="checkbox2"
          type="checkbox"
          value="Valentine's-Day-diy"
        />
        <label
          for="checkbox2"
        >
           Valentine DIY
        </label>
        <br />
        <input
          id="checkbox2"
          name="checkbox3"
          type="checkbox"
          value="Christmas-diy"
        />
        <label
          for="checkbox3"
        >
           Christmas DIY
        </label>
        <br />
        <input
          id="checkbox3"
          name="checkbox4"
          type="checkbox"
          value="Halloween-diy"
        />
        <label
          for="checkbox3"
        >
           Halloween DIY
        </label>
        <br />
        <br />
        <label
          for="dropdownId2"
        >
          <b>
            How did you hear about us? 
          </b>
        </label>
        <div>
          <select
            id="dropdownId2"
            name="dropdown2"
          >
            <option
              value="--"
            >
              ----
            </option>
            <option
              value="SearchEngine"
            >
              Search engine
            </option>
            <option
              value="GoogleAds"
            >
              Google Ads
            </option>
            <option
              value="YoutubeAds"
            >
              Youtube Ads
            </option>
            <option
              value="Twitter post"
            >
              Twitter post
            </option>
            <option
              value="Others"
            >
              Others
            </option>
          </select>
        </div>
        <br />
        <br />
        <button
          type="submit"
        >
          <b>
            Subscribe
          </b>
        </button>
        <br />
        <br />
      </div>
    </form>
  </div>
</div>
`;
